Add detail of responses for the error status codes to Numbers API

lornajane opened the pull request:
# New 

* Describe error responses in Numbers API

# Checklist

- [x] version number incremented (in the `info` section of the spec)

mheap commented on definitions/numbers.yml:
```diff
@@ -58,16 +58,10 @@ paths:
+                $ref: "#/components/schemas/unauthorized"
+            text/xml:
+              schema:
+                $ref: "#/components/schemas/unauthorized"
```
These seem to be missing `xml.name` which controls the name of the wrapper. In the API it's `response` but in the rendered version it's `hash`

lornajane replied:
Good spot. This was already missing from the other endpoints too which I don't think helped but I've added them all now!

lornajane commented:
This was a good catch but it was a bit of a can of worms since most of the responses were missing the XML root tag info! Added for all endpoints, rebased the branch and bumped the version number. This should be in good shape now!

mheap commented:
🙌 Thanks for seeing this through
